### DISCUSSION TITLE
Issue with row selection when autoResetSelectedRows is set to false.

### DISCUSSION THREAD

**Discussion by Ramnath2609** [TanStack/table#3990] [Q&A]
### Describe the bug

I have an initial set of data. Suppose I select row number 3 and 4. And then if the data changes the rows 3 and 4 will still be selected even after data has changed, since autoResetSelectedRows is set to false. Selected rows are maintained by indices. Hence, irrespective of data changes it still selects the particular row index. 

### Your minimal, reproducible example

https://codesandbox.io/s/github/tannerlinsley/react-table/tree/v7/examples/row-selection?file=/src/App.js

### Steps to reproduce

1. Create a table with 10 rows and set autoResetSelectedRows to false.
2. Select row 3 and 4.
3. Now remove 3rd item from data.
4. Now you can see that the selection is different. Since one of my selected row has been removed it should show only one selected row.

### Expected behavior

It will be better if we can provide unique custom ids for selected rows instead of maintaining it as indices.

### How often does this bug happen?

Every time

### Platform

-OS - Windows
-Browser - Google Chrome

### react-table version

v7.7.0

### TypeScript version

v4.4.4

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by abhimanyusingh** [comment]
I am also facing the similar issue. Do we have any solution for this?

**Comment by Sebastiansc** [comment]
same issue workarounds are supper buggy by having to use `useControlledState` to control which indices should be selected after removal
